02.calendar/cal.rb

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env ruby
+
+require 'date' 
+require 'optparse' 
+
+def parse_arguments
+  options = {}
+  OptionParser.new do |opts|
+    opts.on("-y YEAR") do |year|
+      options[:year] = year.to_i
+    end
+    opts.on("-m MONTH") do |month|
+      options[:month] = month.to_i
+    end
+  end.parse! 
+  options 
+end
+
+def print_calendar(year, month)
+  first_date = Date.new(year, month, 1)
+  last_date = Date.new(year, month, -1)
+
+  puts first_date.strftime("%-m月 %Y").center(20)
+  puts "日 月 火 水 木 金 土"
+  print "   " * first_date.wday 
+
+  (first_date..last_date).each do |date|
+    print date.day.to_s.rjust(2)
+    print date.saturday? ? "\n" : " "
+  end
+  print "\n" unless last_date.saturday?
+end
+
+def valid_date?(year, month)
+  year.to_i > 0 && month.to_i.between?(1, 12)
+end
+
+def main
+  options = parse_arguments
+  year = options[:year] || Date.today.year
+  month = options[:month] || Date.today.month
+
+  if valid_date?(year, month)
```

Comment:
```suggestion
  if Date.valid_date?(year, month, 1)
```

Ruby標準のメソッドを使った方がシンプルかも？

https://docs.ruby-lang.org/ja/latest/method/Date/s/valid_civil=3f.html